File: Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/executing/ReplaceMatchesEvaluator.java
package org.opencds.cqf.cql.engine.elm.executing;


public class ReplaceMatchesEvaluator {

    public static Object replaceMatches(String argument, String pattern, String substitution) {
        if (argument == null || pattern == null || substitution == null) {
            return null;
        }

        return argument.replaceAll(pattern, substitution);
    }

}

File: Src/java/engine/src/main/kotlin/org/opencds/cqf/cql/engine/elm/executing/ReplaceMatchesEvaluator.kt
package org.opencds.cqf.cql.engine.elm.executing

object ReplaceMatchesEvaluator {
    @JvmStatic
    fun replaceMatches(argument: String?, pattern: String?, substitution: String?): Any? {
        if (argument == null || pattern == null || substitution == null) {
            return null
        }

        return argument.replace(pattern.toRegex(), substitution)
    }
}
